share/functions/__fish_man_page.fish

```diff
@@ -1,4 +1,16 @@
 function __fish_man_page
-    man (basename (commandline -po; echo)[1]) ^/dev/null
-    or printf \a
+    # Get all commandline tokens not starting with "-"
+    set -l args (string join \n -- (commandline -po) | grep -v '^-')
+
+    # If there are at least two tokens not starting with "-", the second one might be a subcommand.
+    # Try "man first-second" and fall back to "man first" if that doesn't work out.
+    set -l maincmd (basename $args[1])
```

Comment:
This will now print an error when the commandline is empty. 

There's a few ways to stop that, I think I prefer a `set -q args[1]; or return` above.